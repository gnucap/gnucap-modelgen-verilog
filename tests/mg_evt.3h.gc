
attach ./modelgen_1.so
attach ../vams/vsine.so
attach ./d_reject.so


verilog

`modelgen
module test_timer3h(p, n);
	electrical p, n;
	inout p, n;
	(*desc=""*) real cnt, mytime;
	analog begin
		@(timer(mytime)) begin
			$write("evt %g\n", $abstime);
			mytime = $abstime + 1;
			cnt = cnt + 1;
      end
		if(0) $trace("time0 abstime=%g cnt=%g mytime=%g\n", $abstime, cnt, mytime);
	end
endmodule

list

!make test_timer3h.so > /dev/null
attach ./test_timer3h.so

test_timer3h dut0(1,0);
reject #(.time(4.2)) r();

list

print dc v(nodes) iter(0)
print tran v(1) mytime(dut0) iter(0) control(0)
tran 4 basic
dc
tran 6 basic

end
