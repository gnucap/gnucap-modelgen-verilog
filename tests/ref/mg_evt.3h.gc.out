test_timer3h #() dut0 (.p(1),.n(0));
reject #() r ();
#Time       v(1)       mytime(dut0) iter(0)    control(0)
 0.         0.         1.         2.         1.        
 0.04       0.         1.         2.         9.        
evt 1
 1.         0.         2.         2.         2.        
evt 2
 2.         0.         3.         2.         2.        
evt 3
 3.         0.         4.         2.         2.        
evt 4
 4.         0.         5.         2.         1.        
#           v(1)       iter(0)   
 0.         0.         2.        
#Time       v(1)       mytime(dut0) iter(0)    control(0)
 0.         0.         5.         2.         1.        
 0.059524   0.         5.         2.         9.        
 2.5298     0.         5.         4.         17.       
evt 5
 5.         0.         6.         2.         2.        
evt 6
 6.         0.         7.         2.         1.        
